### PR TITLE
Initial docs on terraform

### PR DESCRIPTION
**Pull request by dave2**
Something to note down remote state needs, and basic config like `use_octavia` and `max_retries`.

### CODE REVIEW

File: source/tutorials/orchestration/terraform.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Terraform with teams and CI/CD tools, it may be useful to store state
+remotely. 
+
+Which remote state storage system is best is your choice, however we
+strongly recommend against using the ``swift`` backend. This backend
+has been deprecated by Terraform.
+
+Remote state backends we suggest for Catalyst Cloud are:
+
+* ``pg``, the PostgreSQL backend, which you can easily create using
```
Comment (puck) [inline source/tutorials/orchestration/terraform.rst:84]: Add a link to the service description page on our website or cross reference in the docs here?

File: source/tutorials/orchestration/terraform.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        }
+    }
+
+    provider "openstack" {
+        use_octavia = true
+        max_retries = 10
+    }
+
+In the example above, you may need to change the version number to
```
Comment (puck) [inline source/tutorials/orchestration/terraform.rst:45]: Where do they find the version number to use? https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs ?

File: source/tutorials/orchestration/terraform.rst (hunk 1 above)
Comment (danielobyrne) [reply source/tutorials/orchestration/terraform.rst:84]: Can you expand on the purpose for this link Puck? I'd like to review and make updates to this before merging.

Comment (puck) [reply source/tutorials/orchestration/terraform.rst:84]: A link to https://docs.catalystcloud.nz/database.html

**Comment by dave2** [comment]
I'll fix this in a new MR, and close this one for now.

**Pull request closed without merging**